tests/routes/test_vespasearch.py

```diff
@@ -0,0 +1,743 @@
+
+    db.add(family_document)
+    db.commit()
+    db.add(family_document_slug)
+    db.commit()
+    db.refresh(family_document)
+
+
+def _make_search_request(client, params):
+    response = client.post(SEARCH_ENDPOINT, json=params)
+    assert response.status_code == 200
+    return response.json()
+
+
+@pytest.mark.search
+def test_empty_search_term_performs_browse(client, test_db, mocker):
+    """Make sure that empty search term returns results in browse mode."""
+    _populate_search_db_families(test_db)
+    query_spy = mocker.spy(search._VESPA_CONNECTION, "search")
+
+    body = _make_search_request(client, {"query_string": ""})
+
+    assert body["hits"] > 0
+    assert len(body["families"]) > 0
+    query_spy.assert_not_called()
+
+
+@pytest.mark.search
+def test_simple_pagination_families(test_vespa, client, test_db, monkeypatch):
+    monkeypatch.setattr(search, "_VESPA_CONNECTION", test_vespa)
+    _populate_search_db_families(test_db)
+
+    doc_slugs = []
+    for offset in range(3):
+        params = {
+            "query_string": "and",
+            "limit": 1,
+            "offset": offset,
+        }
+        body = _make_search_request(client, params)
+
+        for f in body["families"]:
+            for d in f["family_documents"]:
+                doc_slugs.append(d["document_slug"])
+
+    assert len(set(doc_slugs)) == len(doc_slugs)
+
+
+@pytest.mark.search
+@pytest.mark.parametrize("exact_match", [True, False])
+def test_search_body_valid(exact_match, test_vespa, client, test_db, monkeypatch):
+    """Test a simple known valid search responds with success."""
+    monkeypatch.setattr(search, "_VESPA_CONNECTION", test_vespa)
+    _populate_search_db_families(test_db)
+
+    body = _make_search_request(
+        client,
+        params={
+            "query_string": "and",
+            "exact_match": exact_match,
+        },
+    )
+
+    fields = sorted(body.keys())
+    assert fields == [
+        "continuation_token",
+        "families",
+        "hits",
+        "query_time_ms",
+        "total_time_ms",
+    ]
+    assert isinstance(body["families"], list)
+
+
+@pytest.mark.search
+@pytest.mark.parametrize("label,query", [("search", "the"), ("browse", "")])
+def test_benchmark_families_search(
+    label, query, test_vespa, monkeypatch, client, test_db
+):
+    monkeypatch.setattr(search, "_VESPA_CONNECTION", test_vespa)
+    _populate_search_db_families(test_db)
+
+    # This is high as it's meant as a last resort for catching new perfomance problems
+    REASONABLE_LATENCY_MS = 25
+
+    times = []
+    for _ in range(1, 10):
+        params = {
+            "query_string": query,
+            "exact_match": True,
+        }
+        body = _make_search_request(client, params)
+
+        time_taken = body["total_time_ms"]
+        times.append(time_taken)
+
+    average = sum(times) / len(times)
+    assert average < REASONABLE_LATENCY_MS
+
+    with open(f"/data/benchmark_{label}_vespa.txt", "w") as out_file:
```

Comment:
How does this work then when thinking about local vs ci? 
If I run tests 4 times during development I'll have my own 4 entries do I commit these or are we adding to a `.gitignore`, then when I push to CI an uncommitted change will be appended as well?

Reply:
All good, everything under the data subfolder is ignored by the gitignore